### PR TITLE
Solution Average-ages

### PR DESCRIPTION
**Pull request by TamaraShukalo**

### CODE REVIEW

File: src/averageAges.js [hunk 1]
```diff
@@ -15,11 +15,16 @@
...
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = people
+    .filter((person) => person.sex === 'm')
+    .filter((person) => !century || Math.ceil(person.died / 100) === century);
```
Comment (tetiana-ivanchenko) [inline src/averageAges.js]:
```suggestion
  const men = people.filter((person) => century 
    ? Math.ceil(person.died / 100) === century && person.sex === 'm'
    : person.sex === 'm'
  );
```

File: src/averageAges.js [hunk 2]
```diff
@@ -37,7 +42,18 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = people
+    .filter((person) => person.sex === 'f')
+    .filter((person) => !withChildren
+      || people
+        .some((potentialMother) => potentialMother.mother === person.name));
```
Comment (tetiana-ivanchenko) [inline src/averageAges.js]: combine filters

File: src/averageAges.js [hunk 3]
```diff
@@ -55,10 +71,22 @@ function calculateWomenAverageAge(people, withChildren) {
...
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  const children = people
+    .filter((child) => child.mother !== undefined)
+    .filter((child) => (!onlyWithSon || child.sex === 'm')
+      && people
```
Comment (tetiana-ivanchenko) [inline src/averageAges.js]: the same

File: src/averageAges.js [hunk 4]
```diff
@@ -37,7 +42,18 @@ function calculateMenAverageAge(people, century) {
...
+    .filter((person) => !withChildren
+      || people
+        .some((potentialMother) => potentialMother.mother === person.name));
+
+  const countWomen = women.length;
+  const totalWomenAge = women.reduce((prev, person) => {
+    return prev + (person.died - person.born);
+  }, 0);
+
+  return totalWomenAge / countWomen;
```
Comment (tetiana-ivanchenko) [inline src/averageAges.js]: move it to the separate function

**Review by tetiana-ivanchenko — CHANGES_REQUESTED** [review]

File: src/averageAges.js [hunk 5]
```diff
@@ -55,10 +71,21 @@ function calculateWomenAverageAge(people, withChildren) {
...
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  const children = people.filter(child => child.mother !== undefined
+    && (!onlyWithSon || child.sex === 'm')
+      && people.some(person => person.name === child.mother)
+  );
```
Comment (anstsot) [inline src/averageAges.js:77]:
```suggestion
  const children = people.filter(child => !!child.mother
    && (!onlyWithSon || child.sex === 'm')
    && people.some(person => person.name === child.mother)
  );
```

**Review by anstsot — APPROVED** [review]
approving:sparkles:

**Pull request closed without merging**